server/channels/store/storetest/user_store.go

```diff
@@ -6182,3 +6186,302 @@ func testUpdateLastLogin(t *testing.T, rctx request.CTX, ss store.Store) {
 	require.NoError(t, err)
 	require.Equal(t, int64(1234567890), user.LastLogin)
 }
+
+func testGetUserReport(t *testing.T, rctx request.CTX, ss store.Store) {
+	now := time.Now()
+
+	u1 := &model.User{Username: "u1" + model.NewId(), DeleteAt: 0}
+	u1.Email = MakeEmail()
+	u1, err := ss.User().Save(u1)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, ss.User().PermanentDelete(u1.Id)) }()
+
+	for i := 0; i < 5; i++ {
+		p := model.Post{UserId: u1.Id, ChannelId: model.NewId(), Message: NewTestId(), CreateAt: now.AddDate(0, 0, -i).UnixMilli()}
+		_, err = ss.Post().Save(&p)
+		require.NoError(t, err)
+	}
+
+	u2 := &model.User{Username: "u2" + model.NewId(), Roles: "system", DeleteAt: 0}
+	u2.Email = MakeEmail()
+	u2, err = ss.User().Save(u2)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, ss.User().PermanentDelete(u2.Id)) }()
+
+	for i := 0; i < 5; i++ {
+		p := model.Post{UserId: u2.Id, ChannelId: model.NewId(), Message: NewTestId(), CreateAt: now.AddDate(0, 0, -i).UnixMilli()}
+		_, err = ss.Post().Save(&p)
+		require.NoError(t, err)
+	}
+
+	u3 := &model.User{Username: "u3" + model.NewId(), Roles: "guest", DeleteAt: now.UnixMilli()}
+	u3.Email = MakeEmail()
+	u3, err = ss.User().Save(u3)
+	require.NoError(t, err)
+
+	team, err := ss.Team().Save(&model.Team{
+		DisplayName: "DisplayName",
+		Name:        NewTestId(),
+		Email:       MakeEmail(),
+		Type:        model.TeamOpen,
+	})
+	require.NoError(t, err)
+	_, err = ss.Team().SaveMember(&model.TeamMember{UserId: u3.Id, TeamId: team.Id}, 1)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, ss.Team().RemoveMember(rctx, team.Id, u3.Id))
+		require.NoError(t, ss.Team().PermanentDelete(team.Id))
+	}()
+
+	defer func() { require.NoError(t, ss.User().PermanentDelete(u3.Id)) }()
+
+	for i := 0; i < 5; i++ {
+		p := model.Post{UserId: u3.Id, ChannelId: model.NewId(), Message: NewTestId(), CreateAt: now.AddDate(0, 0, -i).UnixMilli()}
+		_, err = ss.Post().Save(&p)
+		require.NoError(t, err)
+	}
+
+	err = ss.User().RefreshPostStatsForUsers()
+	require.NoError(t, err)
+
+	t.Run("should return info for all the users", func(t *testing.T) {
+		userReport, err := ss.User().GetUserReport(&model.UserReportOptions{
+			UserReportOptionsWithoutDateRange: model.UserReportOptionsWithoutDateRange{
+				SortColumn: "Username",
+				PageSize:   50,
+			},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, userReport)
+		require.Equal(t, 3, len(userReport))
+
+		require.NotNil(t, userReport[0])
```

Comment:
Are these assertions required even if we do the `len` assertion? (There are similar occurrences below.)

Reply:
I was checking for nil here since it returns a pointer array, was more for safety. Then we check Username for accuracy. I wanted to make sure the tests didn't panic because of that. Is that unnecessary?

Reply:
I ended up changing them to just do the `len`.